src/huggingface_hub/_inference.py

```diff
@@ -0,0 +1,886 @@
+            if value is not None:
+                payload[key] = value
+        response = self.post(json=payload, model=model, task="text-to-image")
+        return _response_to_image(response)
+
+    def text_to_speech(self, text: str, *, model: Optional[str] = None) -> bytes:
+        """
+        Synthesize an audio of a voice pronouncing a given text.
+
+        Args:
+            text (`str`):
+                The text to synthesize.
+            model (`str`, *optional*):
+                The model to use for inference. Can be a model ID hosted on the Hugging Face Hub or a URL to a deployed
+                Inference Endpoint. This parameter overrides the model defined at the instance level. Defaults to None.
+
+        Returns:
+            `bytes`: The generated audio.
+
+        Raises:
+            [`InferenceTimeoutError`]:
+                If the model is unavailable or the request times out.
+            `HTTPError`:
+                If the request fails with an HTTP error status code other than HTTP 503.
+
+        Example:
+        ```py
+        >>> from pathlib import Path
+        >>> from huggingface_hub import InferenceClient
+        >>> client = InferenceClient()
+
+        >>> audio = client.text_to_speech("Hello world")
+        >>> Path("hello_world.wav").write_bytes(audio)
+        ```
+        """
+        response = self.post(json={"inputs": text}, model=model, task="text-to-speech")
+        return response.content
+
+    def _resolve_url(self, model: Optional[str] = None, task: Optional[str] = None) -> str:
+        model = model or self.model
+
+        # If model is already a URL, ignore `task` and return directly
+        if model is not None and (model.startswith("http://") or model.startswith("https://")):
+            return model
+
+        # # If no model but task is set => fetch the recommended one for this task
+        if model is None:
+            if task is None:
+                raise ValueError(
+                    "You must specify at least a model (repo_id or URL) or a task, either when instantiating"
+                    " `InferenceClient` or when making a request."
+                )
+            model = _get_recommended_model(task)
+
+        # Compute InferenceAPI url
+        return (
+            # Feature-extraction and sentence-similarity are the only cases where we handle models with several tasks.
+            f"{INFERENCE_ENDPOINT}/pipeline/{task}/{model}"
+            if task in ("feature-extraction", "sentence-similarity")
+            # Otherwise, we use the default endpoint
+            else f"{INFERENCE_ENDPOINT}/models/{model}"
+        )
+
+
+def _get_recommended_model(task: str) -> str:
+    # TODO: load from a config file? (from the Hub?) Would make sense to make updates easier.
+    if task in RECOMMENDED_MODELS:
+        model = RECOMMENDED_MODELS[task]
+        logger.info(f"Defaulting to recommended model {model} for task {task}.")
```

Comment:
Maybe mention here that users can specify model using `model` param

Reply:
Good idea. Also mentioning that we do not guarantee backward compatibility